app/client/ui/NewRecordButton.ts

```diff
@@ -0,0 +1,54 @@
+import {dom, styled} from 'grainjs';
+import {testId, vars} from 'app/client/ui2018/cssVars';
+import {makeT} from 'app/client/lib/localization';
+import {primaryButton} from 'app/client/ui2018/buttons';
+import {iconSpan} from 'app/client/ui2018/icons';
+import BaseView from 'app/client/components/BaseView';
+
+const t = makeT('NewRecordButton');
+
+const translationStrings = {
+  'record': 'New record',
+  'single': 'New card',
+};
+
+/**
+ * "New Record" button for the given view that inserts a new record at the end on click.
+ *
+ * Note that each view has its own implementation of how to "create a new record"
+ * via the `onNewRecordRequest` method.
+ *
+ * Appears in the bottom-left corner of its parent element.
+ */
+export function newRecordButton(view: BaseView) {
+  const viewType = view.viewSection.parentKey.peek();
+
+  const translationString = translationStrings[viewType as keyof typeof translationStrings]
+    || 'New record';
+  return cssNewRecordButton(
+    iconSpan('Plus'),
+    dom('span', t(translationString)),
+    dom.on('click', () => {
+      if (view.onNewRecordRequest) {
+        view.onNewRecordRequest();
```

Comment:
We've started using the neat new JS/Typescript shorthand: `view.onNewRecordRequest?.()`